### DISCUSSION TITLE
Selecting specific relationship fields

### DISCUSSION THREAD

**Discussion by RoyLee** [mikro-orm/mikro-orm#2594] [Q&A]
**Describe the bug**
Hello, I am trying to use the entity manager to query for specific fields in the relationship data. In my case I am trying to query for all of the author's books but only want the `title` values. Right now I am getting both the `title` and `genre` values. Can you please assist me in figuring out why this is happening?

Author.ts
```
@Entity({ tableName: "authors" })
export class Author extends BaseEntity {
  @Property()
  @Unique()
  name: string;

  @OneToMany('Book', 'author')
  books = new Collection<book>(this);

  constructor(name: string) {
    super();
    this.name = name;
  }
}
```

Book.ts
```
@Entity({ tableName: "books" })
export class Book extends BaseEntity {
  @Property()
  @Unique()
  title: string;

  @Property()
  genre: string;

  @ManyToOne('Author')
  author!: Author;

  constructor(title: string, genre: string) {
    super();
    this.title = title;
    this.genre = genre;
  }
}
```

Code that I am calling:
```
const author = await em().findOne(Author,
  authorId, {
  populate: { books: LoadStrategy.SELECT_IN, },
  fields: ['name', 'books.title'],
});
```

**Expected behavior**
I expect only the book `title` values and not the `genre` values.

**Versions**

| Dependency | Version |
| node | v16.13.1 |
| typescript | v4.3.5 |
| mikro-orm | v4.5.9 |
| mongodb | v4.5.9 |

**Comment by RoyLee** [comment]
One way I can think of doing this is to do 2 separate queries. One to get the author id and then another one to query books for that specific author id. This creates 2 calls to the database which is not ideal.

**Comment by RoyLee** [comment]
Hmm, looks like this is already happening when I used the query in my example. It is running 2 separate queries to the database. Can this be only one?

**Reply by B4nan** [reply]
Nope, in mongo this will always be two queries, there are no joins in mongo (`$lookup` could be used here but it has performance implications which makes it not feasible).

**Comment by RoyLee** [comment]
Just tried running:
```
    const books = await em().findAndCount(Book,
      { author: author.id },
      {
        fields: ['title'],
      });
 ```
 Thought this returns both `genre` and `title` fields. Is my syntax missing anything?

**Reply by B4nan** [reply]
The code is correct, if this does not work for you, you probably already have that entity loaded in your identity map. Which could point to not using request context properly. 

https://mikro-orm.io/docs/identity-map

**Comment by RoyLee** [comment]
Is this issue related to these bugs?
https://github.com/mikro-orm/mikro-orm/issues/1707
https://github.com/mikro-orm/mikro-orm/issues/1739
